File: pinyougou/pinyougou-manage-web/src/main/webapp/js/service/specificationService.js
//定义业务服务
pinyougou.service("specificationService",function ($http) {
    //加载列表数据
    this.findAll = function(){
        return $http.get("../specification/findAll.do");
    };

    this.findPage = function (page, rows) {
        return $http.get("../specification/findPage.do?page=" + page + "&rows=" + rows);
    };

    this.save = function (entity) {
        return $http.post("../specification/save.do",entity);
    };

    this.update = function (entity) {
        return $http.post("../specification/update.do",entity);
    };

    this.findOne = function (id) {
        return $http.get("../specification/findOne.do?id=" + id);
    };

    this.delete = function (ids) {
        return $http.get("../specification/delete.do?ids=" + ids);
    };

    this.search = function (page, rows, searchEntity) {
        return $http.post("../specification/search.do?page=" + page + "&rows=" + rows, searchEntity);

    };
    //查询规格列表
    this.selectOptionList = function () {
        return $http.get("../specification/findSpecification.do");
    };

    /*//查询规格下拉列表
    this.selectOptionList = function () {
        return $http.get("../specification/findSpecification.do");
    }*/

});

File: pinyougou/pinyougou-sellergoods/pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/SpecificationService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbSpecification;
import com.pinyougou.service.BaseService;
import com.pinyougou.vo.PageResult;
import com.pinyougou.vo.Specification;

import java.util.List;
import java.util.Map;

public interface SpecificationService extends BaseService<TbSpecification> {

    PageResult search(Integer page, Integer rows, TbSpecification specification);

    /**
     * 保存规格和规格选项
     * @param specification
     */
    void add(Specification specification);

    /**
     * 查询数据进行回线数据
     */
    Specification findOne(Long id);


    void update(Specification tbSpecification);

    List<Map<String,Object>> findSpecification();
}